File: BinarytoDecimal.swift
import Foundation;

func BinarytoDecimal(_ binaryNumber: Int) -> Int {
	let value: String = String(binaryNumber);
	let len: Int = value.count;
	var output: Int = 0;
	for position in 0...len-1 {
		let char:Character = value[value.index(value.startIndex, offsetBy: position)];
		if (char == "1") {
			output += Int(NSDecimalNumber(value: pow(Double(2), Double(len-1-position))));
		} else if (char == "0") {
			continue;
		} else {
			print("Make sure you only input binary values");
			exit(1);
		}
	}
	return output;
}

print("Binary Number: ", terminator:"");
let input = Int(readLine()!);
print(BinarytoDecimal(input as! Int));

File: BinarytoDecimal.clj
#!/usr/local/bin/clojure -M
(defn binary_to_decimal [binary_number]
    (def value (str binary_number))
    (def len (count value))
    (def output 0)
    (dotimes [position len]
        (def chr (subs value position (+ position 1)))
        (if (= chr "1")
            (def output (+ output (Math/pow 2 (- (- len 1) position))))
            (if (= chr "0")
                (def output (+ output 0))
                (def output "Make sure you only input binary values")
            )
        )
    )
    output
)

(printf "Binary Number: ") (flush)
(printf "%d\n" (int (binary_to_decimal (read-line))))